tensorboard/plugins/text_v2/text_v2_plugin_test.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Integration tests for the Text Plugin."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import collections.abc
+import functools
+import os
+import textwrap
+import numpy as np
+import tensorflow as tf
+
+from tensorboard import plugin_util
+from tensorboard.backend.event_processing import data_provider
+from tensorboard.backend.event_processing import (
+    plugin_event_multiplexer as event_multiplexer,
+)
+from tensorboard.plugins import base_plugin
+from tensorboard.plugins.text_v2 import text_v2_plugin
+from tensorboard.util import test_util
+
+tf.compat.v1.disable_v2_behavior()
+
+GEMS = ["garnet", "amethyst", "pearl", "steven"]
+
+
+class TextPluginTest(tf.test.TestCase):
+    def setUp(self):
+        self.logdir = self.get_temp_dir()
+
+    def create_plugin(self, generate_testdata=True, include_text=True):
+        """Run a test with a `data_provider`."""
+        if generate_testdata:
+            self.generate_testdata(include_text=include_text)
+
+        multiplexer = event_multiplexer.EventMultiplexer()
+        multiplexer.AddRunsFromDirectory(self.logdir)
+        multiplexer.Reload()
+
+        provider = data_provider.MultiplexerDataProvider(
+            multiplexer, self.logdir
+        )
+
+        ctx = base_plugin.TBContext(
+            logdir=self.logdir, multiplexer=multiplexer, data_provider=provider,
+        )
+
+        return text_v2_plugin.TextV2Plugin(ctx)
+
+    def generate_testdata(self, include_text=True, logdir=None):
+        tf.compat.v1.reset_default_graph()
+        sess = tf.compat.v1.Session()
```

Comment:
This is still v1 style. Did you consider using the v2/eager style for generating the data?

Reply:
attempted but has issues when running bazel test (vs just running straight from python)
```
    tf.summary.text("message", message, step)
AttributeError: module 'tensorflow._api.v2.summary' has no attribute 'text'
```

Reply:
fixed, done